packages/protocol-kit/tests/e2e/eip1271.test.ts

```diff
@@ -0,0 +1,475 @@
+      const safeAddress = await safe.getAddress()
+
+      // Adapter and Safe instance for owner 1
+      const ethAdapter1 = await getEthAdapter(account1.signer)
+      const safeSdk1 = await Safe.create({
+        ethAdapter: ethAdapter1,
+        safeAddress,
+        contractNetworks
+      })
+
+      // Adapter and Safe instance for owner 2
+      const ethAdapter2 = await getEthAdapter(account2.signer)
+      const safeSdk2 = await Safe.create({
+        ethAdapter: ethAdapter2,
+        safeAddress,
+        contractNetworks
+      })
+
+      // Adapter and Safe instance for owner 3
+      const safeSdk3 = await Safe.create({
+        ethAdapter: ethAdapter1,
+        safeAddress: signerSafeAddress,
+        contractNetworks
+      })
+
+      return {
+        safe,
+        safeAddress,
+        signerSafe,
+        signerSafeAddress,
+        accounts,
+        contractNetworks,
+        chainId,
+        ethAdapter1,
+        ethAdapter2,
+        safeSdk1,
+        safeSdk2,
+        safeSdk3,
+        fallbackHandlerAddress
+      }
+    })
+
+    itif(safeVersionDeployed >= '1.3.0')(
+      'should validate on-chain messages (Approved hashes)',
+      async () => {
+        const { contractNetworks, safeSdk1, safeSdk2, ethAdapter1 } = await setupTests()
+
+        const chainId = await safeSdk1.getChainId()
+        const safeVersion = await safeSdk1.getContractVersion()
+
+        const customContract = contractNetworks[chainId.toString()]
+
+        const signMessageLibContract = await ethAdapter1.getSignMessageLibContract({
+          safeVersion,
+          customContractAddress: customContract.signMessageLibAddress,
+          customContractAbi: customContract.signMessageLibAbi
+        })
+
+        const messageHash = hashSafeMessage(MESSAGE)
+
+        const txData = signMessageLibContract.encode('signMessage', [messageHash])
+
+        const safeTransactionData: SafeTransactionDataPartial = {
+          to: customContract.signMessageLibAddress,
+          value: '0',
+          data: txData,
+          operation: OperationType.DelegateCall
+        }
+
+        const tx = await safeSdk1.createTransaction({ transactions: [safeTransactionData] })
+        const signedTx = await safeSdk1.signTransaction(tx)
+        const signedTx2 = await safeSdk2.signTransaction(signedTx)
+        const execResponse = await safeSdk1.executeTransaction(signedTx2)
+
+        await waitSafeTxReceipt(execResponse)
+
+        const validatedResponse1 = await safeSdk1.isValidSignature(messageHash)
+        chai.expect(validatedResponse1).to.be.true
+
+        const validatedResponse2 = await safeSdk1.isValidSignature(messageHash, '0x')
+        chai.expect(validatedResponse2).to.be.true
+      }
+    )
+
+    itif(safeVersionDeployed >= '1.3.0')('should revert when message is not signed', async () => {
+      const { safeSdk1 } = await setupTests()
+
+      const response = await safeSdk1.isValidSignature(await hashSafeMessage(MESSAGE), '0x')
```

Comment:
Hash safeMessage not async

```suggestion
      const response = await safeSdk1.isValidSignature(hashSafeMessage(MESSAGE), '0x')
```